### PR TITLE
Clear shard leaders cache when releasing

### PR DESCRIPTION
**Pull request by XuanYang-cn**
See also: #16926

Signed-off-by: yangxuan <xuan.yang@zilliz.com>

/kind bug

### CODE REVIEW

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun ut

File: internal/proxy/meta_cache.go [hunk 1]
```diff
@@ -560,6 +559,25 @@ func (m *MetaCache) GetShards(ctx context.Context, withCache bool, collectionNam
...
+	_, ok := m.collInfo[collectionName]
+	m.mu.RUnlock()
+
+	if !ok {
+		return
+	}
+
+	m.mu.Lock()
+	m.collInfo[collectionName].shardLeaders = nil
+	m.mu.Unlock()
```
Comment (congqixia) [inline internal/proxy/meta_cache.go:582]: Double check is needed if this method and `RemoveCollection` could be called simultaneously.

Comment (XuanYang-cn) [reply internal/proxy/meta_cache.go:582]: ddl task is queued and never will be called concurrently

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**